### PR TITLE
Enable customizeable GM Control Labels

### LINKED ISSUE #574: [release-2.2] Updating ".control.control.envvars[].gm_control_kubernetes_cluster_label" does not update pod labels
The default value for `.control.control.envvars[].gm_control_kubernetes_cluster_label` is `gmproxy`. If a user wants to change this value, it would also need to be reflected in the label key for each pod we expect in the mesh.  However, the pod labels are hardcoded for `gmproxy`.

https://github.com/DecipherNow/helm-charts/blob/da67912805a37c292b5ee64cfeb43e6a14b70534/fabric/control-api/templates/control-api.yaml#L11-L19

### PR DESCRIPTION
**Pull request by chrisbsmith**
**Please provide enough information so that others can review your pull request:**

1. Explain the **details** for making this change. What existing problem does the pull request solve? If it resolves an existing issue, be sure to use a Github [keyword](https://help.github.com/en/articles/closing-issues-using-keywords) to automatically close it.

This update provides the ability to change the pod label that Control uses for k8s service discovery. This updates a lot of files because the `gmproxy` label was hard coded in many places. 

There is a new global configuration variable called `global.control_cluster_label` that can be used to configure this variable. Note that any values with a `.`, `-` or `/` can be used. Prometheus must have these values transformed to `_` to work with the `source_labels`. The chart convert this.

The new default Control label is `greymatter.io/control`

Closes #574 

2. What **changes to global.yaml** are required?

The new `global.control_cluster_label` was added, but it is also defaulted throughout all the charts, so if it is not included in the `global.yaml` it will continue to operate.

3. Have you documented any additional setup steps or configurations options?

The new configuration parameter has been commented.

4. Demonstrate the code is solid. Example: The exact commands you ran and their output, screenshots / videos if the pull request changes UI.

* Install with `make k3d`, `make install`
* Verify the dashboard loads
* Verify the pods are using the correct label
  ```console
  kc get pod data-0 -o json | jq '.metadata.labels'
  {
    "controller-revision-hash": "data-679cf78b86",
    "greymatter.io/control": "data",
    "statefulset.kubernetes.io/pod-name": "data-0"
  }
  ```

### CODE REVIEW

File: data/gm-data/values.yaml [hunk 1]
```diff
@@ -9,6 +9,8 @@ global:
     replicas: 1
   # Port for Grey Matter Control. Used in sidecar envvars
   control_port: 50000
+  # The label Control will use to find pods to include in the mesh
```
Comment (dontlaugh) [inline data/gm-data/values.yaml:12]: Small grammer nit: `s/will use/is used/`. This comment is duplicated a few places, so we ought to update it everywhere.

Comment (chrisbsmith) [reply data/gm-data/values.yaml:12]: So neither of these is probably the best answer.  It's a funny sentence to start with, but I'll update it to read `The label Control uses to find...`.

Comment (dontlaugh) [reply data/gm-data/values.yaml]: ah! That makes more sense now.

**Comment by moosecanswim** [comment]
@chrisbsmith it looks like you did not make the change in `spire/server/templates/statefulset.yaml`

```
   matchLabels:
      app: server
      cluster: spire.server
```

**Comment by chrisbsmith** [comment]
> @chrisbsmith it looks like you did not make the change in `spire/server/templates/statefulset.yaml`
> 
> 
> 
> ```
> 
>    matchLabels:
> 
>       app: server
> 
>       cluster: spire.server
> 
> ```

We don’t add spire to the mesh so the Control label is. It necessary here.

**Comment by moosecanswim** [comment]
Running the following it looks like the only pods that dont have the new label are the init containers and the k3d internal svc pods.


running 
```
#!/bin/bash
pods=$(kubectl get pod | tail -n +2 | awk '{ print $1 }' )
for i in $pods
do
    echo $i
    kubectl get pod $i -o json | jq '.metadata.labels'
done
```
I got:
```
internal-redis-665978b8fb-lbwbn
{
  "deployment": "internal-redis",
  "greymatter.io/control": "internal-redis",
  "pod-template-hash": "665978b8fb"
}
internal-jwt-security-764dbfcd98-cj9r9
{
  "deployment": "internal-jwt-security",
  "greymatter.io/control": "internal-jwt-security",
  "pod-template-hash": "764dbfcd98"
}
redis-5c8d9dcf5d-gx2nq
{
  "deployment": "redis",
  "greymatter.io/control": "redis",
  "pod-template-hash": "5c8d9dcf5d"
}
control-api-0
{
  "controller-revision-hash": "control-api-86c4b498bb",
  "deployment": "control-api",
  "greymatter.io/control": "control-api",
  "statefulset.kubernetes.io/pod-name": "control-api-0"
}
control-api-init-74gtg
{
  "controller-uid": "2856d0b3-042b-4a63-b75a-5906b1131d8c",
  "job-name": "control-api-init"
}
jwt-security-9b869568b-tf4zz
{
  "deployment": "jwt-security",
  "greymatter.io/control": "jwt-security",
  "pod-template-hash": "9b869568b"
}
control-655d9bd5fb-2wlh4
{
  "deployment": "control",
  "greymatter.io/control": "control",
  "pod-template-hash": "655d9bd5fb",
  "run": "control"
}
edge-76b67b5f7d-qx7s4
{
  "deployment": "edge",
  "greymatter.io/control": "edge",
  "pod-template-hash": "76b67b5f7d"
}
svclb-edge-4x58s
{
  "app": "svclb-edge",
  "controller-revision-hash": "8495b58cbc",
  "pod-template-generation": "1",
  "svccontroller.k3s.cattle.io/svcname": "edge"
}
svclb-edge-hn84d
{
  "app": "svclb-edge",
  "controller-revision-hash": "8495b58cbc",
  "pod-template-generation": "1",
  "svccontroller.k3s.cattle.io/svcname": "edge"
}
svclb-edge-lw4n8
{
  "app": "svclb-edge",
  "controller-revision-hash": "8495b58cbc",
  "pod-template-generation": "1",
  "svccontroller.k3s.cattle.io/svcname": "edge"
}
svclb-edge-vd4lw
{
  "app": "svclb-edge",
  "controller-revision-hash": "8495b58cbc",
  "pod-template-generation": "1",
  "svccontroller.k3s.cattle.io/svcname": "edge"
}
svclb-edge-wq8nb
{
  "app": "svclb-edge",
  "controller-revision-hash": "8495b58cbc",
  "pod-template-generation": "1",
  "svccontroller.k3s.cattle.io/svcname": "edge"
}
internal-data-mongo-0
{
  "controller-revision-hash": "internal-data-mongo-6bb7566f5b",
  "greymatter.io/control": "internal-data-mongo",
  "statefulset.kubernetes.io/pod-name": "internal-data-mongo-0"
}
data-mongo-0
{
  "controller-revision-hash": "data-mongo-7586b884dd",
  "greymatter.io/control": "data-mongo",
  "statefulset.kubernetes.io/pod-name": "data-mongo-0"
}
postgres-slo-0
{
  "controller-revision-hash": "postgres-slo-866797fd5b",
  "greymatter.io/control": "postgres-slo",
  "statefulset.kubernetes.io/pod-name": "postgres-slo-0"
}
data-internal-0
{
  "controller-revision-hash": "data-internal-5b4fb666bb",
  "greymatter.io/control": "data-internal",
  "statefulset.kubernetes.io/pod-name": "data-internal-0"
}
data-0
{
  "controller-revision-hash": "data-665f5f9f6f",
  "greymatter.io/control": "data",
  "statefulset.kubernetes.io/pod-name": "data-0"
}
prometheus-0
{
  "controller-revision-hash": "prometheus-677455595",
  "deployment": "prometheus",
  "greymatter.io/control": "prometheus",
  "statefulset.kubernetes.io/pod-name": "prometheus-0"
}
slo-69d4f84854-qwqw6
{
  "deployment": "slo",
  "greymatter.io/control": "slo",
  "pod-template-hash": "69d4f84854"
}
catalog-55d86975f5-c92hq
{
  "deployment": "catalog",
  "greymatter.io/control": "catalog",
  "pod-template-hash": "55d86975f5"
}
dashboard-6f77c6ff67-zlcr2
{
  "greymatter.io/control": "dashboard",
  "pod-template-hash": "6f77c6ff67"
}
catalog-init-zlx2b
{
  "controller-uid": "84999f66-7bc9-42d8-a898-ac50203370c0",
  "job-name": "catalog-init"
}
```

**Comment by chrisbsmith** [comment]
Thanks @moosecanswim. I actually will add the label to those pods so we ensure they are represented as grey matter. Good catch

**Comment by chrisbsmith** [comment]
@moosecanswim Rather than putting the Grey Matter Control label on the spire deployments, I added these labels, which are common with Helm deployments. This will associate the spire deployment with Grey Matter.

When you get a minute can you review again.

```
  labels:
    chart: "{{ $.Chart.Name }}-{{ $.Chart.Version }}"
    release: "{{ $.Release.Name }}"
    heritage: "{{ $.Release.Service }}"
```

**Comment by tilleryd** [comment]
@chrisbsmith LGTM. Everything loaded up in the dashboard.

```
(base)  dtillery@MacBook-Pro  ~/dev/workspace/decipher/gm/helm-charts   pod-labels  kc get pods -o json | jq '.items[].metadata.labels'
{
  "deployment": "internal-redis",
  "greymatter.io/control": "internal-redis",
  "pod-template-hash": "665978b8fb"
}
{
  "deployment": "redis",
  "greymatter.io/control": "redis",
  "pod-template-hash": "5c8d9dcf5d"
}
{
  "deployment": "jwt-security",
  "greymatter.io/control": "jwt-security",
  "pod-template-hash": "8477b49966"
}
{
  "controller-revision-hash": "control-api-54dbbc5c94",
  "deployment": "control-api",
  "greymatter.io/control": "control-api",
  "statefulset.kubernetes.io/pod-name": "control-api-0"
}
{
  "deployment": "internal-jwt-security",
  "greymatter.io/control": "internal-jwt-security",
  "pod-template-hash": "6fffcc84f9"
}
{
  "controller-uid": "933b2b88-15b7-48c1-9ac9-493ad805c014",
  "job-name": "control-api-init"
}
{
  "deployment": "control",
  "greymatter.io/control": "control",
  "pod-template-hash": "78846c4f74",
  "run": "control"
}
{
  "app": "svclb-edge",
  "controller-revision-hash": "755cb575cc",
  "pod-template-generation": "1",
  "svccontroller.k3s.cattle.io/svcname": "edge"
}
{
  "app": "svclb-edge",
  "controller-revision-hash": "755cb575cc",
  "pod-template-generation": "1",
  "svccontroller.k3s.cattle.io/svcname": "edge"
}
{
  "app": "svclb-edge",
  "controller-revision-hash": "755cb575cc",
  "pod-template-generation": "1",
  "svccontroller.k3s.cattle.io/svcname": "edge"
}
{
  "deployment": "edge",
  "greymatter.io/control": "edge",
  "pod-template-hash": "c45988d79"
}
{
  "app": "svclb-edge",
  "controller-revision-hash": "755cb575cc",
  "pod-template-generation": "1",
  "svccontroller.k3s.cattle.io/svcname": "edge"
}
{
  "app": "svclb-edge",
  "controller-revision-hash": "755cb575cc",
  "pod-template-generation": "1",
  "svccontroller.k3s.cattle.io/svcname": "edge"
}
{
  "deployment": "catalog",
  "greymatter.io/control": "catalog",
  "pod-template-hash": "79f87f579d"
}
{
  "controller-revision-hash": "prometheus-54798b85fb",
  "deployment": "prometheus",
  "greymatter.io/control": "prometheus",
  "statefulset.kubernetes.io/pod-name": "prometheus-0"
}
{
  "controller-revision-hash": "postgres-slo-866797fd5b",
  "greymatter.io/control": "postgres-slo",
  "statefulset.kubernetes.io/pod-name": "postgres-slo-0"
}
{
  "controller-revision-hash": "internal-data-mongo-6bb7566f5b",
  "greymatter.io/control": "internal-data-mongo",
  "statefulset.kubernetes.io/pod-name": "internal-data-mongo-0"
}
{
  "controller-revision-hash": "data-mongo-7586b884dd",
  "greymatter.io/control": "data-mongo",
  "statefulset.kubernetes.io/pod-name": "data-mongo-0"
}
{
  "greymatter.io/control": "dashboard",
  "pod-template-hash": "85cd476f7d"
}
{
  "controller-revision-hash": "data-internal-5b7b486f4f",
  "greymatter.io/control": "data-internal",
  "statefulset.kubernetes.io/pod-name": "data-internal-0"
}
{
  "deployment": "slo",
  "greymatter.io/control": "slo",
  "pod-template-hash": "54b4858767"
}
{
  "controller-revision-hash": "data-595bd87c6d",
  "greymatter.io/control": "data",
  "statefulset.kubernetes.io/pod-name": "data-0"
}
{
  "controller-uid": "e5e7a516-f15e-4c30-b43e-be2504973b81",
  "job-name": "catalog-init"
}
```

**Comment by chrisbsmith** [comment]
Thanks @tilleryd !

**Pull request merged**